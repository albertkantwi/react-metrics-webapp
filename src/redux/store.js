import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './country/countrySlice';
import filtersReducer from './searchbox/searchboxSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
    filters: filtersReducer,
  },
});

export default store;
