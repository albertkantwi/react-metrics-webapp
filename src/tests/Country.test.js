import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import CountryDetail from '../components/Country';
import store from '../redux/store';

test('Check if the CountryDetail component is rendered correctly', () => {
  const tree = render(
    <Provider store={store}>
      <BrowserRouter>
        <CountryDetail />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
