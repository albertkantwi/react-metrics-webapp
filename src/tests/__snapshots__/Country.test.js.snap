// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the CountryDetail component is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="country_detail_body"
      >
        <nav
          class="nav_container"
        >
          <a
            class="block_display"
            href="/"
          >
            <img
              alt="Angle Left Arrow"
              class="arrow_logo"
              src="angle-left-solid.svg"
            />
          </a>
          <span
            class="navtext_center"
          >
            Details of
             
             
             
          </span>
          <div
            class="navbox"
          >
            <button
              type="button"
            >
              <img
                alt="Microphone"
                src="microphone-solid.svg"
              />
            </button>
            <button
              type="button"
            >
              <img
                alt="Gear"
                src="gear-solid.svg"
              />
            </button>
          </div>
        </nav>
        <h1
          class="error_back"
        >
          Error, please go back! 
        </h1>
        <div>
          <h3
            class="footer"
          >
            A Microverse Capstone Project by Albert Antwi
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="country_detail_body"
    >
      <nav
        class="nav_container"
      >
        <a
          class="block_display"
          href="/"
        >
          <img
            alt="Angle Left Arrow"
            class="arrow_logo"
            src="angle-left-solid.svg"
          />
        </a>
        <span
          class="navtext_center"
        >
          Details of
           
           
           
        </span>
        <div
          class="navbox"
        >
          <button
            type="button"
          >
            <img
              alt="Microphone"
              src="microphone-solid.svg"
            />
          </button>
          <button
            type="button"
          >
            <img
              alt="Gear"
              src="gear-solid.svg"
            />
          </button>
        </div>
      </nav>
      <h1
        class="error_back"
      >
        Error, please go back! 
      </h1>
      <div>
        <h3
          class="footer"
        >
          A Microverse Capstone Project by Albert Antwi
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
